src/v/cluster/cloud_metadata/offsets_lookup.cc

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cluster/cloud_metadata/offsets_lookup.h"
+
+#include "cluster/cloud_metadata/offsets_lookup_rpc_types.h"
+#include "cluster/logger.h"
+#include "cluster/partition_manager.h"
+#include "cluster/shard_table.h"
+#include "config/configuration.h"
+#include "kafka/server/partition_proxy.h"
+#include "model/ktp.h"
+#include "vlog.h"
+
+#include <seastar/util/later.hh>
+
+namespace cluster::cloud_metadata {
+
+offsets_lookup::offsets_lookup(
+  model::node_id node_id,
+  ss::sharded<cluster::partition_manager>& pm,
+  ss::sharded<cluster::shard_table>& st)
+  : _node_id(node_id)
+  , _partitions(pm)
+  , _shards(st) {}
+
+ss::future<offsets_lookup_reply>
+offsets_lookup::lookup(offsets_lookup_request req) {
+    auto reply = offsets_lookup_reply{};
+    if (req.node_id != _node_id) {
+        // If for some reason we were sent the wrong request, suggest retrying
+        // all NTPs. Perhaps there's a transient metadata issue.
+        reply.node_id = req.node_id;
```

Comment:
is this different than returning an error condition?

Reply:
Not really, it's mostly for logging purposes, albeit confusingly. I'll clean this up and have it return its actual node ID, and log another way.